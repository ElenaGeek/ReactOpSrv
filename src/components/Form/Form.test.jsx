import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

//import Form from './Form'
describe('Form', () => {
    it('render component Form', () => {
      render(<button type="submit" className="btn btn-primary">Submit</button>)
    })
  
    it('render with snapshot', () => {
      const {asFragment} = render(<button type="submit" className="btn btn-primary">Submit</button>)
      expect(asFragment()).toMatchSnapshot()
    })
  
    it('render component with text name Form', () => {
      render(<button type="submit" className="btn btn-primary">Submit</button>)
      expect(screen.getByText(/Submit/)).toBeInTheDocument()
    })
  
    it('button is disable', () => {
      render(<button type="submit" className="btn btn-primary">Submit</button>)
      expect(screen.getByText(/Submit/)).toBeEnabled()
    })

  })