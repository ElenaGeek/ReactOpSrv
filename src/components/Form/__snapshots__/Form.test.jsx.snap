// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary"
    type="submit"
  >
    Submit
  </button>
</DocumentFragment>
`;
